src/lib/Auth.tsx

```diff
@@ -59,7 +61,29 @@ export const Auth: Component = (props) => {
           const { data, error } = await supabase.auth.signUp({
             email: email(),
             password: password(),
+            options: {
+              data: {
+                first_name: firstName(),
+                last_name: lastName(),
+              },
+            },
           });
+          let profileSubmission = {
+            user_id: data.user?.id,
+            first_name: data.user?.user_metadata.first_name,
+            last_name: data.user?.user_metadata.last_name,
+            email: data.user?.email,
+          };
+
+          //Todo: Update RLS to allow anyone to insert into the profile table
+          const { data: profileData, error: profileError } = await supabase
+            .from("profiles")
```

Comment:
This might be wrong but would it be better if you do this with a try catch? I think that it reads more easy with try catch

Reply:
Not sure where you are envisioning the try/catch would go. Not sure I understand the value add here.